app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/AjaxMarkAsRead.php

```diff
@@ -21,17 +44,15 @@ public function execute()
         $notificationId = (int)$this->getRequest()->getPost('id');
         $responseData = [];
         try {
-            $this->_objectManager->create(
-                \Magento\AdminNotification\Model\NotificationService::class
-            )->markAsRead(
-                $notificationId
-            );
+            $this->notificationService->markAsRead($notificationId);
             $responseData['success'] = true;
         } catch (\Exception $e) {
             $responseData['success'] = false;
         }
-        $this->getResponse()->representJson(
-            $this->_objectManager->create(\Magento\Framework\Json\Helper\Data::class)->jsonEncode($responseData)
-        );
+
+        /** @var \Magento\Framework\Controller\Result\Json $resultJson */
+        $resultJson = $this->resultFactory->create(ResultFactory::TYPE_JSON);
+        $resultJson->setData($responseData);
```

Comment:
Calling `setData` on array of response data or calling `setJsonData ` or already formatted data what is "best practice" or does it not matter?